### PR TITLE
feat: Move all nsm dependencies to peer dependencies

### PR DESCRIPTION
**Pull request by razor-x**
A key behavior of this module is generating non-bundled TS code into the callers `.nsm` directory. This code is then consumed by the dependent via their TypeScript compiler. Thus, all imported dependencies must be provided by the consumer.

This update may cause dependency resolution conflicts, so it might be considered a breaking change, however the existing implementation contains implicit dependency requirements which leads to unexpected behavior across any project that does not happen to hoist all of the import dependencies to the root.   

- Update ava and add to peerDependencies
- Update axios and add to peerDependencies
- Add get-port to peerDependencies
- Move debug to peerDependencies
- Move nextjs-middleware-wrappers to peerDependencies
- Set prettier parser option
- Add mime-types to peerDependencies
- Add next to peerDependencies
- Update cookie and add to peerDependencies
- Move raw-body to peerDependencies
- Remove tempy
- Run yarn upgrade
- Update esbuild
- Move content-type to peerDependencies
- Move path-to-regexp to peerDependencies
- Move escape-string-regexp to peerDependencies
- Replace glob-promise with glob 10
- Update node types
- Fix find-up semver string
- Remove unused imports

### CODE REVIEW

**Comment by razor-x** [comment]
@seveibar Found a few other deps to clean up, don't want to sneak past review.

I think this should be merged as a breaking change (major version). It should be simple for consumers to update, but depending on their current dependency versions or package managers they might need to resolve some version conflicts or install missing peer dependencies. If they see those conflicts it would indicate an underlying issue with the version tree though, and it was just "lucky" it all worked before.

After this, we can actually do one more major version to support Next.js v13!

File: package.json [hunk 1]
```diff
@@ -13,10 +13,27 @@
   },
   "license": "MIT",
   "peerDependencies": {
+    "@swc/core": "^1.0.0",
+    "@types/content-type": "^1.0.0",
```
Comment (codetheweb) [inline package.json:17]: are these type packages still required after https://github.com/seamapi/nextjs-server-modules/pull/35?

**Review by codetheweb — APPROVED** [review]
major version bump makes sense to me

File: package.json (hunk 1 above)
Comment (razor-x) [reply package.json:17]: Next was printing warnings about swc, so I had to add it back. It's a good question about the type packages but I think it's safer to keep them for now. If they are really not needed then we can remove them in the future as a patch version.

**Pull request merged**